cpp/src/arrow/vendored/datetime/date.h

```diff
@@ -745,6 +750,65 @@ template<class CharT, class Traits>
 std::basic_ostream<CharT, Traits>&
 operator<<(std::basic_ostream<CharT, Traits>& os, const year_month_day& ymd);
 
+// class year_month_day_pg
+
+class year_month_day_pg {
```

Comment:
Is there a way to simplify this class? The difference from the original `year_month_day` is mainly in `from_days` and `to_days`. Year_month_day can be used as a class template. We can specialize different types of implementations. PostgreSQL's `to_days/from_days` is one of the specializations, and other functions can use the default ones.

Otherwise, there will be a lot of code redundancy when we need to be compatible with the temporal functions of other databases/data management systems in the future.

Reply:
@ZhangHuiGui This class is removed, and I define a enum type named epoch_type used in year_month_day class to calculate the difference value. PTAL, 3Q!

Reply:
I tried to use template class, but more code need to be changed, it's too trouble.